ArduPlane/GCS_Mavlink.cpp

```diff
@@ -1382,22 +1382,20 @@ void GCS_MAVLINK_Plane::handle_set_position_target_global_int(const mavlink_mess
         
         if (pos_target.type_mask & alt_mask)
         {
-            cmd.content.location.alt = pos_target.alt * 100;
-            cmd.content.location.relative_alt = false;
-            cmd.content.location.terrain_alt = false;
+            const int32_t alt_cm = pos_target.alt * 100;
             switch (pos_target.coordinate_frame) 
             {
                 case MAV_FRAME_GLOBAL:
                 case MAV_FRAME_GLOBAL_INT:
+                    cmd.content.location.set_alt_cm(alt_cm, Location::AltFrame::ABSOLUTE);
```

Comment:
If we're playing in here we really should move to `mavlink_coordinate_frame_to_location_alt_frame` instead.

Reply:
Can do!